File: 1980-find-unique-binary-string/1980. Find Unique Binary String.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        n = len(nums)
        nums = set(nums)
        self.res = ""
        def dfs(string):
            if len(string) == n:
                if string not in nums:
                    self.res = string
                return
            
            dfs(string + "1")
            dfs(string + "0")
        dfs("")
        return self.res

File: 1980-find-unique-binary-string/1980. Find Unique Binary String.cpp
class Solution {
public:
    string integerToBinary(int number, int n) {
        string binaryRep = bitset<32>(number).to_string();
        return binaryRep.substr(binaryRep.size() - n);
    }
    string findDifferentBinaryString(vector<string>& nums) {
        int n = nums[0].size();
        int maxVal = (1 << (n)) - 1;
        for (int i = 0; i <= maxVal; i++) {
            string s = integerToBinary(i, n);
            if (find(nums.begin(), nums.end(), s) == nums.end()) {
                return s;
            }
        }
        return "";
    }
};